protob/messages-tron.proto

```diff
@@ -0,0 +1,207 @@
+                optional uint64 frozen_days = 2;            // Days from issue date
+            }
+            optional bytes owner_address = 1;               // Account address
+            optional bytes name = 2;                        // Asset name
+            optional bytes abbr = 3;                        // Asset abbreviation
+            optional uint64 total_supply = 4;               // Total supply including frozen
+            repeated TronFrozenSupply frozen_supply = 5;    // Frozen supply
+            optional uint32 trx_num = 6;                    // Amount of TRX (exchange ratio)
+            optional uint32 num = 8;                        // Amount of tokens (exchange ratio)
+            optional uint64 start_time = 9;                 // Negotiation start date and time
+            optional uint64 end_time = 10;                  // Negotiation end date and time
+            optional bytes description = 20;                // Asset description
+            optional bytes url = 21;                        // Asset URL
+        }
+        // Participate in an asset 
+        message TronParticipateAssetIssueContract {
+            optional bytes owner_address = 1;               // Account address
+            optional bytes to_address = 2;                  // Asset issuer address
+            optional bytes asset_name = 3;                  // The name of target asset
+            optional uint64 amount = 4;                     // TRX amount in sun
+        }
+        // Freeze TRX balance                        
+        message TronFreezeBalanceContract {
+            optional bytes owner_address = 1;               // Account address
+            optional uint64 frozen_balance = 2;             // Amount to freeze
+            optional uint64 frozen_duration = 3;            // Freeze minimal duration in days
+        }
+        // Unfreeze TRX Balance
+        message TronUnfreezeBalanceContract {
+            optional bytes owner_address = 1;               // Account address
+        }
+        // Unfreeze Asset Balance
+        message TronUnfreezeAssetContract {
+            optional bytes owner_address = 1;               // Account address
+        }
+        // Withdraw witness balance
+        message TronWithdrawBalanceContract {
+            optional bytes owner_address = 1;               // Account address
+        }
+        // Update Asset
+        message TronUpdateAssetContract {
+            optional bytes owner_address = 1;               // Account address
+            optional bytes description = 2;                 // New description
+            optional bytes url = 3;                         // New URL
+        }
+        // Network proposal contract
+        message TronProposalCreateContract {
+            optional bytes owner_address = 1;               // Account address
+            map<uint64, uint64> parameters = 2;             // Parameter to be changed
```

Comment:
we currently don't support implicit maps, and we probably don't want to; please convert to explicit repeated "(name, value)" submessage.

Reply:
OK, updated.